src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/PollLogsStrategy.cs

```diff
@@ -97,18 +115,76 @@ public void Notify(IStorageBlob blobWritten)
+
+            _tryEnqueue = 0;
+            _skippedEnqueue = 0;
+
+            long[] readTimes = new long[_logListeners.Count];
+            int forLoopCount = 0;
             // Poll the logs (to detect ongoing writes).
             foreach (BlobLogListener logListener in _logListeners.Values)
             {
-                cancellationToken.ThrowIfCancellationRequested();
-
-                foreach (IStorageBlob blob in await logListener.GetRecentBlobWritesAsync(cancellationToken))
+                int index = forLoopCount++;
+                scanLogs.Add(Task.Run(async () =>
                 {
                     cancellationToken.ThrowIfCancellationRequested();
-                    await NotifyRegistrationsAsync(blob, cancellationToken);
+
+                    long before = _stopWatch.ElapsedMilliseconds;
+                    var blobs = await logListener.GetRecentBlobWritesAsync(cancellationToken);
+                    readTimes[index] = _stopWatch.ElapsedMilliseconds - before;
+
+                    List<Task> enqueues = new List<Task>();
+                    foreach (IStorageBlob blob in blobs)
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+                        enqueues.Add(NotifyRegistrationsAsync(blob, failedNotifications, cancellationToken));
+                    }
+                    await Task.WhenAll(enqueues);
+
+                }));
+            }
+
+            await Task.WhenAll(scanLogs);
+            _stopWatch.Stop();
+            int logTryEnqueue = _tryEnqueue;
+            int logSkipped = _skippedEnqueue;
+            int logFailed = failedNotifications.Count - beforeLogFailed;
+            int logSuccess = logTryEnqueue - logSkipped - logFailed;
+            long logEnqueueTime = _stopWatch.ElapsedMilliseconds - beforeLogTime;
+            long averageReadTime = (long)readTimes.Average();
+
+            // Re-add any failed notifications for the next iteration.
+            foreach (IStorageBlob failedNotification in failedNotifications)
+            {
+                _blobsFoundFromScanOrNotification.Enqueue(failedNotification);
+            }
+
+            if (_trace != null)
+            {
+                // output metrics, assume trace method is thread 
+                if (beforeLogTryEnqueue > 0)
+                {
+                    _traceString.AppendLine($"log inventory ==> Enqueue: {beforeLogTryEnqueue} messages, {beforeLogSuccess} successes, {beforeLogSkipped} skipped. Cost: {beforeLogTime}ms");
+                }
+                if (logTryEnqueue > 0)
+                {
+                    _traceString.AppendLine($"log new ==> Enqueue: {logTryEnqueue} messages, {logSuccess} successes, {logSkipped} skipped. Average log cost: {averageReadTime}ms, total cost: {logEnqueueTime}ms");
+                }
+                if (_traceString.Length > 0)
+                {
+                    _trace.Verbose(_traceString.ToString());
```

Comment:
I'm concerned about the noise these logs will introduce. I realize you've made them Verbose, but they're really beyond that from the customer's point of view - they're Debug/Diagnostic level. We have customers who have set their logs to Verbose and I don't think they'd want to see these logs.

Recommend separating out your tracing proposal from the parallelization work since they are separate, so we can decide on what to do there separately.